### PR TITLE
Implement warning when viewing old guides version

### PR DESCRIPTION
**Pull request by SaladFork**
_Addresses #506. A rebase of @acorncom's #563 which rebased @wagenet's #114. cc @michaelrkn_

##### Todo:
 - ~~How does `'CURRENT_REVISION_WILL_APPEAR_HERE_WHEN_DEPLOYED'` (in `version-select.js.erb`) work and how/why does it differ from `'<%= data.search.revision %>'` (in `versions.js.erb`)?~~
 - ~~How does `versions.json` get set when deploying?~~
 - ~~Would be nice to have documentation for the above~~
 - [x] Remove changes to `versions.json` and `version-select.js.erb` as we won't be supporting beta versions ([todo source](https://github.com/emberjs/guides/pull/563#issuecomment-129709603))
 - [x] Update code to handle `versions.json` format of `["v1.10.0", "v1.11.0", "v1.12.0", "v1.13.0"]`. ([todo source](https://github.com/emberjs/guides/pull/563#issuecomment-129709603))
 - [x] Clean up duplicate AJAX calls to `version.json` in `version-select.js.erb` and `outdated-warnings.js` ([todo source](https://github.com/emberjs/guides/pull/563#issuecomment-129709603))
 - [x] Add mechanism to allow user to jump straight to latest guide
   - Do we want to do this with an actionable button ("Jump to latest guides"), an informative link ("The latest guides are for vx.x.x"), or otherwise?
 - [x] Squash rebase prior to merge

##### Issues (advice welcome):
 - [x] Currently `outdated-warnings.js` and `version-select.js.erb` get concat'd before `versions.js.erb` (at least when running locally). This causes an issue as `window.GUIDE_VERSIONS` hasn't been defined yet.

### CODE REVIEW

**Comment by SaladFork** [comment]
cc @michaelrkn

File: source/javascripts/app/guides/versions.js.erb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+window.GUIDE_VERSIONS = {
+  current: '<%= data.search.revision %>',
```
Comment (michaelrkn) [inline source/javascripts/app/guides/versions.js.erb]: we no longer have this `data` property. instead, use `available[0]`.

**Comment by michaelrkn** [comment]
Awesome work!

> How does 'CURRENT_REVISION_WILL_APPEAR_HERE_WHEN_DEPLOYED' (in version-select.js.erb) work?
> How does versions.json get set when deploying?

From https://github.com/michaelrkn/guides.emberjs.com/blob/versions/tasks/update-versions.js.

> how/why does 'CURRENT_REVISION_WILL_APPEAR_HERE_WHEN_DEPLOYED' differ from '<%= data.search.revision %>' (in versions.js.erb)

`data.search.revision` is no longer used.

> Add mechanism to allow user to jump straight to latest guide

The version switcher replaces the version number in the URL with the most recently released version (see https://github.com/emberjs/guides/blob/master/source/javascripts/app/version-select.js.erb) to try to keep the user on the same page when switching versions. If we want to add this feature, then we should extract that logic into something that can be used in both version switching and the outdated warning feature.

**Comment by michaelrkn** [comment]
Can you move `versions.js` and `outdated-warnings.js.erb` out of the `guides` folder and just into the `app` folder?

**Comment by michaelrkn** [comment]
Am I right in thinking this fixes #600?

**Comment by SaladFork** [comment]
> Am I right in thinking this fixes #600?

Yes:
```javascript
// Sort versions and make latest appear first
this.available = available.sort(compareVersions).reverse();
```

**Comment by SaladFork** [comment]
> Can you move `versions.js` and `outdated-warnings.js.erb` out of the guides folder and just into the app folder?

Sure, though as-implemented this warning won't appear on any page outside the guides, even if it may be outdated (such as API docs).

**Comment by michaelrkn** [comment]
The `guides` folder was a vestige of the extraction of the guides from the main website. If you take a look at the directory structure on master, it's not there any more.

**Comment by locks** [comment]
Status?

**Comment by SaladFork** [comment]
@locks: this is at the top of my OSS todo list :). Planning on getting to it this weekend.

**Comment by SaladFork** [comment]
- [x] Use `available[0]` instead of `<%= data.search.revision %>`
 - [ ] Use same mechanism for dropdown version switch as button in warning version switch
 - [x] Move `versions.js` and `outdated-warnings.js.erb` out of the guides folder

**Comment by SaladFork** [comment]
Pushed updates. Please feel free to suggest changes to the wording / button. Otherwise it should be good to go.

How it looks now:

**Comment by locks** [comment]
nitpick, can you make it `VIEW v2.0.0` instead?

**Comment by locks** [comment]
@michaelrkn do we want squashed commits here?

**Comment by SaladFork** [comment]
@locks: Done.



Planning on squashing as soon as I get a :+1: (see last remaining item in OP).

**Comment by michaelrkn** [comment]
:+1: :)

**Comment by michaelrkn** [comment]
Thanks for all your hard work on this @SaladFork!

**Comment by acorncom** [comment]
@SaladFork Agreed.  I got it re-started, but then ran out of time to complete.  Thanks for pushing it the rest of the way! :+1:

**Pull request merged**